src/main/java/io/boomerang/eventing/nats/jetstream/PubTransmitter.java

```diff
@@ -80,32 +87,114 @@ public void publish(String subject, String message)
-        .build();
-    // @formatter:on
+  // TODO replace this w one line where needed instead of a method call?
+  private void storeMessage(String subject, String message) {
+    System.out.println("Storing message to send later");
+    failedMessages.add(createNATSMessage(subject, message));
+    printObject(failedMessages);
+  }
 
-    // Publish the message
-    PublishAck publishAck = connection.jetStream().publish(natsMessage);
+  // TODO replace this w one line where needed instead of a method call?
+  private NatsMessage createNATSMessage(String subject, String message) {
+    return NatsMessage.builder().subject(subject).data(message, StandardCharsets.UTF_8).build();
+  }
+
+  private void publishFailedMessages() {
+    // removed unnecessary arguments from this method
+    // convert failedMessages List<Message> -> ConcurrentLinkedQueue<Message>
+    // remove one message at a time after successful publish
+    // attempt actions inside synchronized while loop,
+    // break if queue is empty or if connection fails again
+    
+    while (true) {
+      synchronized (this) {
+        try {
+          System.out.println("Resending failed message");
+          PublishAck publishAck =
+              connectionPrimer.getActiveConnection().jetStream().publish(failedMessages.peek());
+          System.out.println("PublishAck:");
+          printObject(publishAck);
+          if (publishAck != null) {
+            System.out.println("-------------------------------------------------");
+            System.out.println("Queue size before: " + failedMessages.size());
+            printObject(failedMessages);
+            
+            System.out.println("Removing published message");
+            failedMessages.poll();
+            
+            System.out.println("Queue size after: " + failedMessages.size());
+            printObject(failedMessages);
+            System.out.println("-------------------------------------------------");
+          }
+        } catch (IOException | JetStreamApiException e) {
+          e.printStackTrace();
+        }
+
+        if (failedMessages == null || connectionPrimer.getActiveConnection() == null) {
+          break;
+        }
+      }
+    }
+  }
 
-    logger.debug("Message published to the stream! " + publishAck);
+  // TODO remove this after testing, also remove dependency from pom.xml
+  private void printObject(Object object) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    try {
+      System.out.println(objectMapper.writeValueAsString(object));
+    } catch (JsonProcessingException e) {
+      // log an error
+    }
+  }
+
+  @Override
+  public void connectionUpdated(ConnectionPrimer connectionPrimer) {
+    // TODO don't retry messages if failed stream
+    Connection connection = connectionPrimer.getActiveConnection();
+    if (connection != null) {
+      System.out.println("Connection active");
+      try {
+        getJetStream(connection);
```

Comment:
If this fails due to no Stream being present on the server (based on the provided configuration) and `automaticallyCreateStream` is set to `false`, there is no reason to re-publish the messages, since these will fail anyway.
The code here should fail silently (for now) and you must clear out the queue.